addons/udes_stock_picking_batch/models/stock_picking.py

```diff
@@ -1,5 +1,55 @@
-from odoo import models
+from odoo import models, api
 
 
 class StockPicking(models.Model):
     _inherit = "stock.picking"
+
+    @api.constrains("batch_id")
+    def _trigger_batch_confirm_and_remove(self):
+        """
+       Batch may require new pickings to be auto confirmed or removed
+       Note: this function is very fragile, any change might brake its functionality
+        """
+        orig_batches = self.env.context.get("orig_batches")
+        if orig_batches:
+            for batch in orig_batches:
+                batch._assign_picks()
+                batch._remove_unready_picks()
+                batch._compute_state()
+        for batch in self.batch_id:
+            batch._assign_picks()
+            batch._remove_unready_picks()
+            batch._compute_state()
+
+    @api.constrains("state")
+    def _trigger_batch_state_recompute(self):
+        """
+        Changes to picking state cause batch state recompute, may also cause
+        unready pickings to be removed from the batch
+        Note: this function is very fragile, any change might brake its functionality
+        """
+        orig_batches = self.env.context.get("orig_batches")
+        if orig_batches:
+            for batch in orig_batches:
+                batch._remove_unready_picks()
+                batch._compute_state()
+        for batch in self.batch_id:
+            batch._remove_unready_picks()
+            batch._compute_state()
+
+    def action_assign(self):
+        """
+        Recompute batch state. In theory this is not necessary
+        but the constraint on state has not proven to work correctly.
+        """
+        super(StockPicking, self).action_assign()
+        self.batch_id._compute_state()
+
+    def write(self, vals):
+        """If writing picking, check if previous batch is now complete"""
+        # This will be used to trigger recompute of the batch state
+        # we can't relate on the state after as batch_id might be
+        # removed in the write
+        batches = self.batch_id
+        context_vals = {"orig_batches": batches} if batches else {}
```

Comment:
```
if batches:
    self= self.with_context(orig_batches=batches)
return super(StockPicking, self).write(vals)
```

I think looks cleaner and no need to use kwargs for a single context variable

Reply:
Addressed